tensorflow/core/transforms/remapper/remapping_helper.h

```diff
@@ -0,0 +1,260 @@
+    Type dtype;
+    if (auto attr = contraction_op->getAttrOfType<TypeAttr>(attr_name)) {
+      dtype = attr.getValue();
+    } else {
+      return false;
+    }
+    if (dialect_->IsConv2D(contraction_op)) {
+      return dtype.isa<Float32Type>();
+    } else if (dialect_->IsMatMul(contraction_op)) {
+      return dtype.isa<Float32Type, Float64Type>();
+    } else {
+      return false;
+    }
+  }
+
+  // This function is currently used by contraction ops.
+  bool IsCpuCompatibleDataType(Operation* contraction_op,
+                               const StringRef& attr_name = "T") {
+    Type dtype;
+    if (auto attr = contraction_op->getAttrOfType<TypeAttr>(attr_name)) {
+      dtype = attr.getValue();
+    } else {
+      return false;
+    }
+
+    if (is_onednn_enabled_) {
+      // Only contraction ops (MatMul, Conv2D, Conv3D, and
+      // DepthwiseConv2dNative) and BatchMatMul are supported. BatchMatMul
+      // fusions are handled differently than contraction ops.
+      bool is_supported = IsContraction(contraction_op) ||
+                          dialect_->IsAnyBatchMatMul(contraction_op);
+      return is_supported && dtype.isa<Float32Type, BFloat16Type>();
+    }
+
+    if (dialect_->IsConv2D(contraction_op)) {
+      return dtype.isa<Float32Type, Float64Type>();
+    } else if (dialect_->IsMatMul(contraction_op)) {
+      return dtype.isa<Float32Type>();
+    } else {
+      return false;
+    }
+  }
+
+  // This function is currently used by convolution type op
+  bool IsGpuCompatibleDataFormat(Operation* conv_op,
+                                 const StringRef& attr_name = "data_format") {
+    StringRef data_format;
+    if (conv_op->hasAttrOfType<StringAttr>(attr_name)) {
+      data_format = conv_op->getAttrOfType<StringAttr>(attr_name).getValue();
+    } else {
+      return false;
+    }
+    if (dialect_->IsConv2D(conv_op)) {
+      return data_format == "NHWC" || data_format == "NCHW";
+    } else {
+      return false;
+    }
+  }
+
+  // This function is currently used by convolution type op
+  bool IsCpuCompatibleDataFormat(Operation* conv_op,
+                                 const StringRef& attr_name = "data_format") {
+    StringRef data_format;
+    if (conv_op->hasAttrOfType<StringAttr>(attr_name)) {
+      data_format = conv_op->getAttrOfType<StringAttr>(attr_name).getValue();
+    } else {
+      return false;
+    }
+    if (dialect_->IsConv2D(conv_op)) {
+      return data_format == "NHWC" ||
+             (is_onednn_enabled_ && data_format == "NCHW");
+    } else if (dialect_->IsConv3D(conv_op)) {
+      return data_format == "NDHWC" ||
+             (is_onednn_enabled_ && data_format == "NCDHW");
+    } else {
+      return false;
+    }
+  }
+
+  bool IsGpuCompatible(const ContractionBiasAddActivation& pattern) {
+#if TENSORFLOW_USE_ROCM
+    // ROCm does not support _FusedConv2D. Does it suppport _FusedMatMul?
+    return false;
+#endif
```

Comment:
Is there any utility function for this? like IsMklEnabled()? I would prefer something like that if possible

Reply:
I am not aware of such function. We will include this once have it.